packages/nitro-protocol/contracts/interfaces/IForceMoveApp.sol

```diff
@@ -27,3 +27,40 @@ interface IForceMoveApp {
         uint256 nParticipants
     ) external pure returns (bool);
 }
+
+/**
+ * @dev The IForceMoveApp interface calls for its children to implement an application-specific validTransition function, defining the state machine of a ForceMove state channel DApp.
+ */
+interface IForceMoveApp2 {
+    struct VariablePart {
+        bytes outcome;
+        bytes appData;
+    }
+
+    /**
+     * @notice Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @dev Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @param a State being transitioned from.
+     * @param b State being transitioned to.
+     * @param turnNumB Turn number being transitioned to.
+     * @param nParticipants Number of participants in this state channel.
+     * @param signedBy Bit field showing which participants signed state b.
```

Comment:
This `validTransition` function is awkward. It is hard (impossible?) to support things like:
- From an ABC state, Alice can move to an A state, and Bob can move to a B state
- From an A state, Bob can move to an AB state
- From a B state, Alice can move ton an AB state

Reply:
I wasn't sure what you were driving at here until the extra context of #3552.